docs/types.rst

```diff
@@ -902,6 +908,23 @@ possible:
 It is planned to remove this restriction in the future but currently creates
 some complications because of how arrays are passed in the ABI.
 
+.. index:: !inline;arrays
+
+Inline Arrays
+^^^^^^^^^^^^^
+
+You can initialise a statically sized memory array inline using syntax such as ``string[] memory myarray = new string[](4);``. For example::
```

Comment:
Should we be very specific and note that currently this does not work with dynamically sized arrays?